packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java

```diff
@@ -304,7 +307,12 @@ private void queryPurchasesAsync(String skuType, MethodChannel.Result result) {
     if (billingClientError(result)) {
       return;
     }
+    postQueryPurchasesOnMainThread(skuType, result, new Handler(Looper.getMainLooper()));
+  }
 
+  @VisibleForTesting
+  public void postQueryPurchasesOnMainThread(
```

Comment:
Isn't in the response, not the query, that is happening on the main thread? The name seems misleading.

Reply:
And actually, since you're taking the handler as a parameter, nothing in this method is enforcing that it's the main thread AFAICT. If the only valid thing to pass as a handle is a main-thread handler, that should definitely be called out in a declaration comment.

I assume it's being passed instead of created internally for unit testing purposes? If so, couldn't we instead just create the handler internally, and use a mock `Result` that asserts that its calls are happening on the main thread?